packages/react-instantsearch/examples/autocomplete/src/App-Hits-And-Facets.js

```diff
@@ -0,0 +1,143 @@
+import React, {Component} from 'react';
+import {InstantSearch} from 'react-instantsearch/dom';
+import {createConnector} from 'react-instantsearch';
+import {connectSearchBox, connectRefinementList} from 'react-instantsearch/connectors';
+import Autosuggest from 'react-autosuggest';
+import {isEqual, forOwn} from 'lodash';
+/* eslint-disable import/no-unresolved */
+import 'react-instantsearch-theme-algolia/style.css';
+/* eslint-enable import/no-unresolved */
+
+class App extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {searchState: {}, hits: {}, query: ''};
+  }
+
+  onSearchStateChange(searchState) {
+    this.setState({searchState: {...this.state.searchState, ...searchState}});
+  }
+
+  onProps(props) {
+    props.facets[props.indexName] = props.hits;
+    this.setState({hits: {...this.state.hits, ...props.facets}, query: props.query});
+  }
+
+  formatHitsForAutoSuggest() {
+    const hits = [];
+    forOwn(this.state.hits, (value, key) => {
+      hits.push({title: key, hits: value});
+    });
+    return hits;
+  }
+
+  render() {
+    return (
+      <div>
+        <FirstResults onProps={this.onProps.bind(this)}
```

Comment:
You should bind that in the constructor so it doesn't create a new function at each `render`.